pandas/core/arrays/datetimes.py

```diff
@@ -221,9 +221,7 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
 
         result = cls._simple_new(values, freq=freq, tz=tz)
         if freq_infer:
-            inferred = result.inferred_freq
-            if inferred:
-                result.freq = to_offset(inferred)
+            result.freq = to_offset(result.inferred_freq)
```

Comment:
Just curious, why can we drop the `if`-logic?

Reply:
because it is unnecessary.  if `inferred` is `None`, then we're setting `result.freq=None` which is what it should be anyway.  It isn't a big deal, but these 4 lines are repeated 4 times and after looking at these for weeks I find it an eyesore.